src/impl/transact_log_handler.cpp

```diff
@@ -27,10 +27,168 @@
 #include <realm/lang_bind_helper.hpp>
 
 #include <algorithm>
+#include <numeric>
 
 using namespace realm;
 
 namespace {
+
+class KVOAdapter : public _impl::TransactionChangeInfo {
+public:
+    KVOAdapter(std::vector<BindingContext::ObserverState>& observers, BindingContext* context);
+
+    void before(SharedGroup& sg);
+    void after(SharedGroup& sg);
+
+private:
+    BindingContext* m_context;
+    std::vector<BindingContext::ObserverState>& m_observers;
+    std::vector<void *> m_invalidated;
+
+    struct ListInfo {
+        BindingContext::ObserverState* observer;
+        size_t col;
+        _impl::CollectionChangeBuilder builder;
+    };
+    std::vector<ListInfo> m_lists;
+    VersionID m_version;
+};
+
+KVOAdapter::KVOAdapter(std::vector<BindingContext::ObserverState>& observers, BindingContext* context)
+: m_context(context)
+, m_observers(observers)
+{
+    if (m_observers.empty())
+        return;
+
+    std::vector<size_t> tables_needed;
+    for (auto& observer : observers) {
+        tables_needed.push_back(observer.table_ndx);
+    }
+    std::sort(begin(tables_needed), end(tables_needed));
+    tables_needed.erase(std::unique(begin(tables_needed), end(tables_needed)), end(tables_needed));
+
+    auto realm = context->realm.lock();
+    auto& group = realm->read_group();
+    for (auto& observer : observers) {
+        auto table = group.get_table(observer.table_ndx);
+        for (size_t i = 0, count = table->get_column_count(); i < count; ++i) {
+            if (table->get_column_type(i) != type_LinkList)
+                continue;
+            m_lists.push_back({&observer, i, {}});
+        }
+    }
+
+    auto max = max_element(begin(tables_needed), end(tables_needed));
```

Comment:
Is ADL guaranteed to find `std::max_element` here? The iterator types seem like they could plausibly be raw pointer types, which aren't necessarily in the `std` namespace.